java/flight/flight-core/src/main/java/org/apache/arrow/flight/DictionaryUtils.java

```diff
@@ -66,8 +68,9 @@ static Schema generateSchemaMessages(final Schema originalSchema, final FlightDe
           count);
       final VectorUnloader unloader = new VectorUnloader(dictRoot);
       try (final ArrowDictionaryBatch dictionaryBatch = new ArrowDictionaryBatch(
-          id, unloader.getRecordBatch())) {
-        messageCallback.accept(new ArrowMessage(dictionaryBatch));
+          id, unloader.getRecordBatch());
+          final ArrowMessage message = new ArrowMessage(dictionaryBatch)) {
```

Comment:
should this message be closed?

Reply:
This message is in a try-with-finally so it will be closed. The callback is responsible for incrementing the reference count if it does need to hold on to a reference (and indeed, this will eventually call asInputStream which handles all of that).